metrics-operator/main.go

```diff
@@ -93,6 +101,25 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	if SLIFilePath != "" {
```

Comment:
let's move that part into a separate function

Reply:
that won't make much difference, we would still need to call that function, which return values would be `(string, error)` and there will still need to be the block checking the err != nil and afterwards writing out the resulting string.

I mean, we can always move everything into a separate function, but in this case, we don't gain anything

Reply:
can we not just do something like the following?

```go
if SLIFilePath != "" {
  // at this point we know we'll return in any case, so no need to 
  // make the function return a (string, error) here
  return convertSLI(SLIFilePath, provider, namespace)
}

// ...

func convertSLI(SLIFilePath, provider, namespace string) {
                //read file content
 		fileContent, err := os.ReadFile(SLIFilePath)
 		if err != nil {
 			log.Fatalf("error reading file content: %s", err.Error())
 			return
 		}
 		// convert
 		c := converter.NewSLIConverter()
 		content, err := c.Convert(fileContent, provider, namespace)
 		if err != nil {
 			log.Fatalf(err.Error())
 			return
 		}
 		// write out converted result
 		fmt.Print(content)
 		return
}
```

That to me does look cleaner

Reply:
I updated it in a slightly different way, I don't think writing out output to stdout in a non-main function is a good practice